lib/core-classes/DecodersService.ts

```diff
@@ -1,31 +1,70 @@
-import { PluginContext, EmbeddedSDK, Plugin } from "kuzzle";
+import { DecoderQueue } from './../types/decoders/decodersQueue';
+import { PluginContext, EmbeddedSDK, Plugin, Inflector, PluginImplementationError, PluginApiDefinition } from "kuzzle";
 
 import { Decoder } from "./Decoder";
+import { PayloadHandler, PayloadService } from './PayloadService';
 import { DeviceManagerConfig } from "../DeviceManagerPlugin";
-import { DecoderContent } from '../types/decoders/DecodersContent';
+import { DecoderContent } from "../types/decoders/DecodersContent";
 
 export class DecodersService {
   private config: DeviceManagerConfig;
   private context: PluginContext;
+  private api: PluginApiDefinition;
+  private payloadService: PayloadService;
 
   private decoders: Map<string, Decoder>;
 
   get sdk(): EmbeddedSDK {
     return this.context.accessors.sdk;
   }
 
-  constructor(plugin: Plugin, decoders: Map<string, Decoder>) {
+  constructor(plugin: Plugin, decoders: Map<string, Decoder>, decodersQueue: DecoderQueue[], payloadService: PayloadService) {
     this.config = plugin.config as any;
     this.context = plugin.context;
+    this.api = plugin.api;
+    this.payloadService = payloadService;
 
     this.decoders = decoders;
+
+    decodersQueue.forEach(({ decoder, handler }) => {
```

Comment:
Try to avoid `forEach` loops because they uses function which gave no control on the execution flow.

For example this will not wait for each decoder but rather run all the functions in //
```
      decoders.forEach(async decoder => {
        await someFunction();
        this.register(decoder);        
      })
```